support/cas-server-support-rest-authentication/src/main/java/org/apereo/cas/adaptors/rest/RestAuthenticationHandler.java

```diff
@@ -77,6 +85,31 @@ protected AuthenticationHandlerExecutionResult authenticateUsernamePasswordInter
         }
         throw new FailedLoginException("Rest endpoint returned an unknown response for " + c.getUsername());
     }
+
+    /**
+     * Resolve {@link MessageDescriptor warnings} from the {@link ResponseEntity authenticationResponse}.
+     *
+     * @param authenticationResponse The response sent by the REST authentication endpoint
+     * @return The warnings for the created {@link AuthenticationHandlerExecutionResult}
+     */
+    protected List<MessageDescriptor> getWarnings(final ResponseEntity<?> authenticationResponse) {
+        List<MessageDescriptor> messageDescriptors = new ArrayList<>();
+
+        var passwordExpirationDate = authenticationResponse.getHeaders().getFirstZonedDateTime("X-Cas-PasswordExpirationDate");
```

Comment:
use `val`.

Reply:
Your checkstyle rules allow both `lombok.val` and `var` :https://github.com/apereo/cas/blob/3d6092ffc210a312b8eaa8b081429ffe743fb568/style/checkstyle-rules.xml#L549

Why do you prefer `lombok.val` over the java-native `var` keyword?